### ISSUE TITLE
Memory issues with making Fixture is a global member

### ISSUE THREAD

**Issue by marcusnaslund** [magic-lang/ooc-kean#1584]
Only one major obstacle stands in the way of removing all implicit `this`/`This`. In every `expect` call we use the static property `is` defined in `Fixture` as:

```ooc
is ::= static IsConstraints new()
```

When moving this line to global scope, `is := IsConstraints new()`, tests always crash. This is because, when a test gets freed, the `free` call propagates up to freeing `is` which then gets doubly freed if a test has more than one `expect` call. This can be "fixed" by adding

```ooc
free: override func
```

to `IsConstraints` in `source/unit/Constraints.ooc` which prevents it from getting freed (the `super` call never happens). Exactly why this works when `is` is a static property of `Fixture` I do not know, but someone well-versed in ooc/rock might have a better answer.

However, this approach still does not work with the concurrent tests (but seems to work fine with all others). Running `./test.sh concurrent` with a global `is`, even when `free` has been changed, results in all kinds of `SIGSEGV`s and `SIGABRT`s. 

It might be that our use of `is` has some memory and/or thread-safety issues, which might be the reason for the uncommon (but still existing) problems on upstream with the `BlockedQueue` and `ThreadPool` tests (#1523), as well as random memory problems with `clang` and under Windows (#1466).

Long story short, this needs a closer examination. Ideas and inputs are very welcome @thomasfanell @fredrikbryntesson @sebastianbaginski @davidpiuva

**Comment by marcusnaslund** [comment]
We might solve this by passing `null` instead of `this` as the parent argument when creating constraints from `Is`... I will investigate it tomorrow.

**Comment by sebastianbaginski** [comment]
What if `is` was just a property, not a static one ? This object seems quite cheap to create.

**Comment by davidpiuva** [comment]
Will concurrency problems go away if we just give a boolean result to expect?

**Comment by marcusnaslund** [comment]
> What if is was just a property, not a static one ? This object seems quite cheap to create.

Sure, but it should also not be necessary. I will experiment some more today after #1582 is merged.

> Will concurrency problems go away if we just give a boolean result to expect?

No idea, but that's not an acceptable compromise, the main idea of `Fixture` is to provide good error messages if/when a test fails.
